ReleaseTooling/Sources/ZipBuilder/FrameworkBuilder.swift

```diff
@@ -346,34 +346,23 @@ struct FrameworkBuilder {
         "\(error)")
     }
 
-    // Find the location of the public headers, any platform will do.
     guard let anyPlatform = targetPlatforms.first,
           let archivePath = slicedFrameworks[anyPlatform] else {
       fatalError("Could not get a path to an archive to fetch headers in \(frameworkName).")
     }
 
-    // Get the framework Headers directory. On macOS, it's a symbolic link.
-    let headersDir = archivePath.appendingPathComponent("Headers").resolvingSymlinksInPath()
-
-    // The macOS Headers directory can have a Headers file in it symbolically linked to nowhere.
-    // Delete it here to avoid putting it in the zip or crashing the Carthage hash generation.
-    // For example,in the 8.0.0 zip distribution see
-    // Firebase/FirebaseAnalytics/PromisesObjC.xcframework/macos-arm64_x86_64/PromisesObjc
-    //  .framework/Headers/Headers
-    do {
-      try fileManager.removeItem(at: headersDir.appendingPathComponent("Headers"))
-    } catch {
-      // Ignore
-    }
-
     // Find CocoaPods generated umbrella header.
     var umbrellaHeader = ""
+    // TODO(ncooke3): Evaluate if `TensorFlowLiteObjC` is needed?
     if framework == "gRPC-Core" || framework == "TensorFlowLiteObjC" {
       // TODO: Proper handling of podspec-specified module.modulemap files with customized umbrella
       // headers. This is good enough for Firebase since it doesn't need these modules.
+      // TODO(ncooke3): Is this needed for gRPC-Core?
```

Comment:
@paulb777 do you recall the context behind this TODO?

Reply:
I mean the:
```
 // TODO: Proper handling of podspec-specified module.modulemap files with customized umbrella
 // headers. This is good enough for Firebase since it doesn't need these modules.
```

Reply:
It looks like doing something different for gRPC-Core is still needed: 
https://github.com/grpc/grpc/blob/8bab8f1724c4049224b95ef85d319bdf56ddb83e/gRPC-Core.podspec#L680-L69

Reply:
What about TensorFlowLiteObjC? I see it is mentioned a few times throughout the source, is that still needed?

Reply:
Probably still needed. Historically, the Release Builder did the wrong thing for podspecs with `s.module_map`. Also, there was conflation between pod name and module name that required workarounds like from https://github.com/grpc/grpc/blob/8bab8f1724c4049224b95ef85d319bdf56ddb83e/gRPC-Core.podspec#L54.  I don't remember the deeper level details. We once got `TensorFlowLiteObjC` working so maybe worth keeping in case it matters in the future.

Reply:
Thank you for the context!